### PR TITLE
commonlib: Add AbstractHostFilePlugin

### PR DESCRIPTION
**Pull request by kingthorin**
- Updated commonlib to target 2.10 more specifically (now that 2.10 has been released).
- Added AbstractHostFilePlugin and UnitTest.
- Updated other commonlib classes to use Log4j 2.x infrastructure.
- Updated ElmanScanRule to leverage AbstractHostFilePlugin.

Fixes zaproxy/zaproxy#6133

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by thc202** [comment]
The add-on `ascanrules` needs to depend on the new version of `commonlib`.

**Comment by kingthorin** [comment]
Dependency version set.

**Comment by kingthorin** [comment]
Further genericized the commonlib addition, and added a maint note to it's CHANGELOG.

**Comment by kingthorin** [comment]
De-conflicted.

File: addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/AbstractHostFilePlugin.java [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+                    getName(),
+                    e.getClass().getName(),
+                    e.getMessage());
+            return;
+        }
+        if (isFalsePositive(newRequest)) {
+            return;
+        }
+        int statusCode = newRequest.getResponseHeader().getStatusCode();
+        if (statusCode == HttpStatusCode.OK || isPage200(newRequest)) {
```
Comment (thc202) [inline addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/AbstractHostFilePlugin.java:183]: Explicitly checking the status code defeats the custom page definition, maybe use just `isSuccess`?

File: addOns/commonlib/src/test/java/org/zaproxy/addon/commonlib/AbstractHostFilePluginUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Plugin.AlertThreshold;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.testutils.ActiveScannerTestUtils;
+import org.zaproxy.zap.testutils.NanoServerHandler;
+
+/** Unit test for {@link AbstractHostFilePlugin}. */
+public abstract class AbstractHostFilePluginUnitTest<T extends AbstractHostFilePlugin>
```
Comment (thc202) [inline addOns/commonlib/src/test/java/org/zaproxy/addon/commonlib/AbstractHostFilePluginUnitTest.java:43]: This one is not actually being used(?).

Comment (kingthorin) [reply addOns/commonlib/src/test/java/org/zaproxy/addon/commonlib/AbstractHostFilePluginUnitTest.java:43]: I’ll have to look at the other examples, I probably missed something.

**Comment by kingthorin** [comment]
Fixed.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**